prow/plugins/golint/golintsuggestion/golint_suggestion.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package golintsuggestion
+
+import (
+	"regexp"
+	"strings"
+	"github.com/golang/lint"
+)
+
+const (
+	lintNamesRegex1 = "don't use underscores in Go names; (.*) should be (.*)"
+	nameUnderscore = "NAME_UNDERSCORE"
+	lintNamesRegex2 = "don't use ALL_CAPS in Go names; use CamelCase"
+	nameAllCaps = "NAME_ALL_CAPS"
+)
+
+var allCapsRE = regexp.MustCompile(`.+[A-Z0-9_]+.+$`)
+
+var lintCaseMap = map[string]string {
+	lintNamesRegex1: nameUnderscore,
+	lintNamesRegex2: nameAllCaps,
+}
+
+// CanSuggestCodeChange determines whether a code suggestion can be made
+// for a given lint.Problem
+func CanSuggestCodeChange(p lint.Problem) bool {
+	for r := range lintCaseMap {
+		result, _ := regexp.MatchString(r, p.Text)
+		if result {
+			return true
+		}
+	}
+	return false
+}
+
+// SuggestCodeChange returns code suggestions for a given lint.Problem
+// Returns empty string if no suggestion can be given
+func SuggestCodeChange(p lint.Problem) string {
+	if !CanSuggestCodeChange(p) {
+		return ""
+	}
+	var caseName = ""
+	var matchedRegex = ""
+	for r, c := range lintCaseMap {
+		result, _ := regexp.MatchString(r, p.Text)
+	 	if result {
+			caseName = c
+			matchedRegex = r
+	 	}
+	}
+	if caseName == "" {
+		return ""
+	}
+
+	var result string
+	switch caseName {
+	case nameUnderscore:
```

Comment:
Instead of mapping from regexps to string keys, `lintCaseMap` should map from regexps to handler functions that generate the suggestion text for the appropriate problem type.
That would get rid of this switch statement and move each case into its own function for better readability and unit testing.